### PR TITLE
remove ambiguity detection test

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- This test doesn't have enough value for now, and creates a lot of merge conflicts. This is painful when keeping PRs up to date, and when cherry picking for a version

## Solution

- Remove this test. Readd it when we reach 0 ambiguities

### CODE REVIEW

File: tests/ecs/ambiguity_detection.rs [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-    );
-
-    // RenderApp is not checked here, because it is not within the App at this point.
-    let render_extract_ambiguities = count_ambiguities(app.sub_app(RenderExtractApp));
-    assert_eq!(
-        render_extract_ambiguities.total(),
-        0,
-        "RenderExtract app has unexpected ambiguities among the following schedules: \n{:#?}",
-        render_extract_ambiguities,
-    );
```
Comment (Vrixyz) [inline tests/ecs/ambiguity_detection.rs:48]: This is at 0 ; so we might as well just keep this, to be sure it doesn't regress ?

I'd argue to keep the rest commented out, to be slightly more straightforward to test, but only print the total and remove the assert?

Another idea would be to compare ambiguity count with master, and offer a non blocking warning if the count is different, but it's slightly more work.

**Comment by urben1680** [comment]
For context, does the count `44` goes both up or down with PRs? If it only goes down, it could be a `<= 44` check and each bevy release could set the number to the current number.

**Comment by Vrixyz** [comment]
> it could be a `<= 44` check

I'd approve that :D

For even more context: the initial objective of setting `==` was to avoid removing one but adding another one, which could be as bad as an increasing count, but the resulting conflicts are probably not worth it.

**Comment by alice-i-cecile** [comment]
This test is *incredibly* valuable to allow us to monotonically reduce the number of ambiguities present. Without it in place, changes have repeatedly regressed it from zero ambiguities, causing a number of real bugs, both large and small. Trying to remove these all in a single PR is extremely hard to review, and quite a bit of work.

I would grudgingly accept a change to swap `==` to `<=`, and would welcome any improvements to the error output though. I would also suggest removing this test for now when preparing point releases.

That said, I'll prioritize getting these resolved: I don't like the current state of things for contributors either.

**Pull request closed without merging**